src/Plugin/Field/FieldFormatter/AddThisBasicButtonFormatter.php

```diff
@@ -52,11 +52,11 @@ public function settingsForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items) {
-    $settings = $this->getSettings();
-    $add_this_service = \Drupal::service('addthis.addthis');
-    $markup = $add_this_service->getBasicButtonMarkup($settings);
+
     return array(
-      '#markup' => $markup
+      '#type' => 'addthis_basic_button',
+      '#size' => $config['basic_button']['button_size'],
```

Comment:
$config does not exist in the context of this function. Blocks use $this->configuration while Fields use $this->getSettings(). You will need to adjust these for the fields.